reader.go

```diff
@@ -261,3 +261,24 @@ func (r *Reader) WriteTo(w io.Writer) (n int64, err error) {
 		}
 	}
 }
+
+// CheckLZ4 returns a bool indicating if the first 32 bytes mactch LZ4 header
+// from the io.Reader stream.
+//
+// False if it could not determine, e.g. error in reading from io.Reader.
```

Comment:
if the underlying io.Reader returns an error, the error should be passed through.

Reply:
it is passed through now. PTAL. this comment no longer applies. deleted